shell/src/main/java/alluxio/cli/LogLevel.java

```diff
@@ -0,0 +1,257 @@
+  private static final Options OPTIONS = new Options()
+      .addOption(TATGET_OPTION)
+      .addOption(LOG_NAME_OPTION)
+      .addOption(LEVEL_OPTION);
+
+  /**
+   * Prints the help message.
+   *
+   * @param message message before standard usage information
+   */
+  public static void printHelp(String message) {
+    System.err.println(message);
+    HelpFormatter help = new HelpFormatter();
+    help.printHelp(LOG_LEVEL, OPTIONS, true);
+  }
+
+  /**
+   * Implements log level setting and getting.
+   *
+   * @param args list of arguments contains target, logName and level
+   * @return 0 on success, 1 on failures
+   * @exception ParseException if there is an error in parsing
+   * @exception IOException if the rpc server cannot reached
+   */
+  public static int logLevel(String... args) throws ParseException, IOException {
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd = parser.parse(OPTIONS, args, true /* stopAtNonOption */);
+
+    List<TargetInfo> targets = parseOptTarget(cmd);
+    String logName = parseOptLogName(cmd);
+    String level = parseOptLevel(cmd);
+
+    for (TargetInfo targetInfo : targets) {
+      if (handleTarget(targetInfo, logName, level)) {
+        return 1;
+      }
+    }
+    return 0;
+  }
+
+  private static List<TargetInfo> parseOptTarget(CommandLine cmd) throws IOException {
+    String[] targets;
+    if (cmd.hasOption(TARGET_OPTION_NAME)) {
+      String argTarget = cmd.getOptionValue(TARGET_OPTION_NAME);
+      if (StringUtils.isBlank(argTarget)) {
+        targets = new String[]{"master", "workers"};
+      } else if (argTarget.contains(TARGET_SEPARATOR)) {
+        targets = argTarget.split(TARGET_SEPARATOR);
+      } else {
+        targets = new String[]{argTarget};
+      }
+    } else {
+      targets = new String[]{"master", "workers"};
+    }
+    return getTargetInfos(targets);
+  }
+
+  private static List<TargetInfo> getTargetInfos(String[] targets) throws IOException {
+    List<TargetInfo> targetInfoList = new ArrayList<>();
+    for (String target : targets) {
+      if (target.equals("master")) {
+        String masterHost = NetworkAddressUtils.getConnectHost(ServiceType.MASTER_WEB);
+        int masterPort = NetworkAddressUtils.getPort(ServiceType.MASTER_WEB);
+        targetInfoList.add(new TargetInfo(masterHost, masterPort, "master"));
+      } else if (target.equals("workers")) {
+        AlluxioBlockStore alluxioBlockStore = AlluxioBlockStore.create();
+        List<BlockWorkerInfo> workerInfoList = alluxioBlockStore.getWorkerInfoList();
+        for (BlockWorkerInfo workerInfo : workerInfoList) {
+          WorkerNetAddress netAddress = workerInfo.getNetAddress();
+          targetInfoList.add(
+              new TargetInfo(netAddress.getHost(), netAddress.getWebPort(), "worker"));
+        }
+      } else if (target.contains(":")) {
+        String[] hostPortPair = target.split(":");
+        int port = Integer.parseInt(hostPortPair[1]);
```

Comment:
There are some edge cases, how about changing the `if` condition to `target.split(":").length >= 2`?
```java
":".split(":").length = 0
"host:".split(":").length = 1
":port".split(":").length = 1
```